jdat_notebooks/preimaging/preimaging_01_mirage.ipynb

```diff
@@ -0,0 +1,515 @@
+    "        doc = yaml.load(stream, Loader=yaml.FullLoader)\n",
+    "        \n",
+    "        filtname = doc['Readout']['filter'] #read the filter keyword\n",
+    "        if not os.path.exists(os.path.join(path,'yaml',filtname.lower())):\n",
+    "            os.mkdir(os.path.join(path,'yaml',filtname.lower()))\n",
+    "    \n",
+    "    filetomove = yamlfile  \n",
+    "    input_file = filetomove\n",
+    "    output_file = os.path.join(path,'yaml',filtname.lower()) \n",
+    "    \n",
+    "    print('input  = ',input_file)\n",
+    "    print('output = ',output_file)\n",
+    "    \n",
+    "    shutil.move(input_file, output_file) #move the file to the corresponding sub directory\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Execute MIRAGE and create simulated data\n",
+    "Now that the yaml files have been generated, we can execute MIRAGE using them as input parameters and generate the NIRCam images."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As an example, let us choose filter F444W. We are going to simulate all the images that were observed using filter F444W. The variable \"listname\" contains the names of the yaml files that we want to process through MIRAGE. There are 32 F444W yaml files.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# input parameters\n",
+    "listname = 'values-f444w.txt'   # text file with list of yaml files \n",
+    "filtname = 'f444w'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# copy the F444W yaml files back in the working directory\n",
+    "cwd = os.getcwd()\n",
+    "filter_pattern = os.path.join(cwd,'yaml',filtname.lower(),'jw*yaml') \n",
+    "files = glob(filter_pattern)\n",
+    "\n",
+    "for yamlfile in files:\n",
+    "    input_file = yamlfile         \n",
+    "    output_file = cwd \n",
+    "    print('input  = ',input_file)\n",
+    "    print('output = ',output_file)\n",
+    "    shutil.copy(input_file, output_file) #this copies over filter files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# read the list of yaml files to process\n",
+    "t = Table.read(listname, format='ascii.fast_no_header')\n",
+    "input_yaml = t['col1']\n",
+    "\n",
+    "yaml_list = []\n",
+    "for k in range(len(input_yaml)):\n",
+    "    yaml_list.append(input_yaml[k])\n",
+    "\n",
+    "print(yaml_list)\n",
+    "\n",
+    "files = yaml_list\n",
+    "paramlist = yaml_list\n",
+    "print(files)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# here we loop over the yaml files and run MIRAGE in a sequence\n",
```

Comment:
Maybe add a note saying that from each yaml file, Mirage will produce a noiseless seed image, a "raw" [(level 1b) file](https://jwst-pipeline.readthedocs.io/en/stable/jwst/data_products/science_products.html?highlight=uncal#uncalibrated-raw-data-uncal), and a linearized ramp (equivalent to the output of the linearity correction step of the [calwebb_detector1 pipeline](https://jwst-pipeline.readthedocs.io/en/stable/jwst/pipeline/calwebb_detector1.html))